### PR TITLE
feat(search): basic FT.ALTER

### LINKED ISSUE #3142: Search: FT.ALTER
Quickest implementation: full index rebuild

### PR DESCRIPTION
**Pull request by dranikpg**
Fixes #3142

### CODE REVIEW

File: src/server/search/search_family.cc [hunk 1]
```diff
@@ -469,6 +469,53 @@ void SearchFamily::FtCreate(CmdArgList args, ConnectionContext* cntx) {
   cntx->SendOk();
 }
 
+void SearchFamily::FtAlter(CmdArgList args, ConnectionContext* cntx) {
+  CmdArgParser parser{args};
+  string_view idx_name = parser.Next();
+  parser.ExpectTag("SCHEMA");
```
Comment (romange) [inline src/server/search/search_family.cc:475]: you should check parser for errors

File: src/server/search/search_family.cc [hunk 2]
```diff
@@ -469,6 +469,53 @@ void SearchFamily::FtCreate(CmdArgList args, ConnectionContext* cntx) {
...
+      },
+      false);
+
+  if (!index_info) {
+    cntx->transaction->Conclude();
+    return cntx->SendError("Index not found");
+  }
+
+  // Parse additional schema
+  optional<search::Schema> new_fields = ParseSchemaOrReply(index_info->type, parser, cntx);
```
Comment (romange) [inline src/server/search/search_family.cc:496]: it is not possible to remove fields?

File: src/server/search/search_family.cc (hunk 1 above)
Comment (dranikpg) [reply src/server/search/search_family.cc:475]: Right 🤓

File: src/server/search/search_family.cc (hunk 2 above)
Comment (dranikpg) [reply src/server/search/search_family.cc:496]: I was surprised, but no, it only has SCHEMA ADD 🤷🏻‍♂️

File: src/server/search/search_family.cc [hunk 3]
```diff
@@ -469,6 +469,59 @@ void SearchFamily::FtCreate(CmdArgList args, ConnectionContext* cntx) {
...
+  schema.field_names.insert(new_fields->field_names.begin(), new_fields->field_names.end());
+
+  // Rebuild index
+  // TODO: Introduce partial rebuild
+  auto upd_cb = [idx_name, index_info](Transaction* tx, EngineShard* es) {
+    es->search_indices()->DropIndex(idx_name);
+    es->search_indices()->InitIndex(tx->GetOpArgs(es), idx_name, index_info);
+    return OpStatus::OK;
+  };
+  cntx->transaction->Execute(upd_cb, true);
```
[the remark this answers is not in the document]
Comment (romange) [reply src/server/search/search_family.cc:520]: not now

**Review by romange — APPROVED** [review]

**Pull request merged**